lib/Dialect/FIRRTL/Transforms/InferResets.cpp

```diff
@@ -582,7 +582,7 @@ ResetSignal InferResetsPass::guessRoot(ResetNetwork net) {
 // the element type is uniform across all elements.
 
 static unsigned getMaxFieldID(FIRRTLBaseType type) {
-  return TypeSwitch<FIRRTLBaseType, unsigned>(type)
+  return TypeSwitch<FIRRTLBaseType, unsigned>(type.getAnonymousType())
```

Comment:
Aren't these functions duplicates of stuff in Utils?

Reply:
Yes, I opened an issue before but still not fixed yet https://github.com/llvm/circt/issues/4760